crates/metrics/src/config.rs

```diff
@@ -0,0 +1,81 @@
+use itertools::Itertools;
+use once_cell::sync::Lazy;
+use strum::IntoEnumIterator;
+use strum_macros::{
+    Display,
+    EnumIter,
+    EnumString,
+};
+
+#[derive(Debug, Display, Clone, Copy, PartialEq, EnumString, EnumIter)]
+#[strum(serialize_all = "lowercase")]
+pub enum Module {
+    Importer,
+    P2P,
+    Producer,
+    TxPool, /* TODO[RC]: Not used. Add support in https://github.com/FuelLabs/fuel-core/pull/2321 */
+    GraphQL, // TODO[RC]: Not used... yet.
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct Config(Vec<Module>);
+impl Config {
+    pub fn is_enabled(&self, module: Module) -> bool {
+        !self.0.contains(&module)
+    }
+}
+
+impl std::convert::From<&str> for Config {
+    fn from(s: &str) -> Self {
+        if s == "all" {
+            return Self(Module::iter().collect())
+        }
+        Self(
+            s.split(',')
+                .filter_map(|s| s.parse::<Module>().ok())
+                .collect(),
+        )
+    }
+}
+
+static HELP_STRING: Lazy<String> = Lazy::new(|| {
+    format!(
+        "Comma-separated list of modules or 'all' to disable all metrics. Available options: {}, all",
+        Module::iter().join(", ")
+    )
+});
+
+pub fn help_string() -> &'static str {
+    &HELP_STRING
+}
+
+#[cfg(test)]
+mod tests {
+    use strum::IntoEnumIterator;
+
+    use crate::config::{
+        Config,
+        Module,
+    };
+
+    #[test]
```

Comment:
can we add a test for invalid values?

Reply:
Added in https://github.com/FuelLabs/fuel-core/pull/2310/commits/1b5c428b13b875cfd318414c4d5ffb760fad2971